src/main/java/spinbox/Parser.java

```diff
@@ -43,98 +45,164 @@ public static void setPageTrace(ArrayDeque<String> pageTraceNew) {
     }
 
     /**
-     * Builds the required page data for command input. Return page and maybe moduleCode
+     * Appends the page or page and "modules" from pageTrace.
+     * @param pageData The original empty page data.
+     * @param tempPageTrace The cloned version of the original ArrayDeque pageTrace.
+     * @param lastElement The last element of the ArrayDeque pageTrace.
+     * @return The full page data.
+     */
+    private static String emptyPageComponentAppender(String pageData, ArrayDeque<String> tempPageTrace,
+                                                     String lastElement) {
+        String fullPageData;
+        fullPageData = pageData.concat(lastElement);
+        tempPageTrace.removeLast();
+        if (tempPageTrace.size() != 0) {
+            String lastComponent = tempPageTrace.getLast();
+            String finalFullPageData;
+            finalFullPageData = fullPageData.concat(" " + lastComponent.toUpperCase());
```

Comment:
Might want to declare + initialize in one line

Reply:
Okay I have fixed this.